Modules/selectmodule.c

```diff
@@ -94,24 +101,37 @@ typedef struct {
 } pylist;
 
 static void
-reap_obj(pylist fd2obj[FD_SETSIZE + 1])
+reap_obj(pylist *fd2obj, size_t setsize)
 {
-    unsigned int i;
-    for (i = 0; i < (unsigned int)FD_SETSIZE + 1 && fd2obj[i].sentinel >= 0; i++) {
+    size_t i;
+    for (i = 0; i < setsize + 1 && fd2obj[i].sentinel >= 0; i++) {
         Py_CLEAR(fd2obj[i].obj);
     }
     fd2obj[0].sentinel = -1;
 }
 
 
+/* returns the size of a given Python sequence
+   returns -1 if the input is not a sequence
+*/
+static Py_ssize_t
+seqsize(PyObject *seq)
+{
+    PyObject* fast_seq = PySequence_Fast(seq, "argument 1 must be sequence");
```

Comment:
What is the benefit of using `PySequence_Fast()` here instead of `PySequence_Length()` or even `PyObject_Length()`?
(Despite the fact that you missed `Py_DECREF(fast_seq)`)

Reply:
It's just my first PR to python - so I'm not really familiar with the code-base. I just did what I saw in other functions in the same file. But I'll update it - PySequence_Length/PyObject_Length look promising.